etl/src/main/java/org/pih/hivmigration/etl/sql/TreatmentObsMigrator.groovy

```diff
@@ -0,0 +1,327 @@
+              comments varchar(4000),
+              KEY `source_encounter_idx` (`source_encounter_id`)
+            );
+        ''')
+
+        loadFromOracleToMySql('''
+            INSERT INTO hivmigration_ordered_other (source_encounter_id, ordered, comments)
+            VALUES (?, ?, ?)
+        ''', '''
+            SELECT encounter_id, ordered, comments
+            FROM hiv_ordered_other
+        ''')
+
+        executeMysql("CALL create_tmp_obs_table();")
+        setAutoIncrement("tmp_obs", "(SELECT max(obs_id) + 1 FROM obs)")
+
+
+        // doesn't appear in the OpenMRS UI but is meaningful data
+        executeMysql("Add Prophylaxis Not Indicated to tmp obs table", '''
+            INSERT INTO tmp_obs (source_encounter_id, concept_uuid, value_coded_uuid)
+            SELECT
+                source_encounter_id,
+                concept_uuid_from_mapping('PIH', '2740'),
+                concept_uuid_from_mapping('CIEL', '1066')
+            FROM hivmigration_observations
+            WHERE observation = 'prophylaxis_not_indicated' AND value = 't';
+        ''')
+
+        executeMysql("Add Prophylaxis: Yes", '''
+            INSERT INTO tmp_obs (source_encounter_id, concept_uuid, value_coded_uuid)
+            SELECT
+                source_encounter_id,
+                concept_uuid_from_mapping('PIH', '2740'),
+                concept_uuid_from_mapping('CIEL', '1065')
+            FROM hivmigration_ordered_other
+            WHERE ordered IN (
+                'tmp_smx_prophylaxis', 'inh_prophylaxis', 'fluconazole', 'mosquito_net', 'other_prophylaxis',
+                'tmp_smx_prophylaxis_continue', 'inh_prophylaxis_continue', 'fluconazole_continue', 'mosquito_net_continue', 'other_prophylaxis_continue'
+            );
+        ''')
+        // TODO: include old prophylaxes
+
+        //
+        // BEGIN prophylaxis group PIH:10742
+
+        executeMysql("Create intermediate table for prophylaxis rows", '''
+            CREATE TABLE hivmigration_prophylaxis (
+                obs_group_id int primary key auto_increment,
+                name varchar(80),
+                start boolean,
+                cont boolean,
+                stopped boolean,
+                other_value varchar(80),
+                source_encounter_id int,
+                encounter_date date
+            );
+        ''')
+
+        setAutoIncrement('hivmigration_prophylaxis', '(select max(obs_id)+1 from tmp_obs)')
+
+        executeMysql("Load intermediate prophylaxis table", '''
+            DROP PROCEDURE IF EXISTS load_prophylaxis;
+            DELIMITER $$ ;
+            CREATE PROCEDURE load_prophylaxis(_txName varchar(80))
+            BEGIN
+                INSERT INTO hivmigration_prophylaxis
+                    (name, start, cont, stopped, other_value, source_encounter_id, encounter_date)
+                SELECT
+                    _txName,
+                    IF(MAX(base.ordered = _txName), TRUE, FALSE) AS start,
```

Comment:
Can you explain what these "MAX" functions are doing?  Doesn't seem right.

Reply:
Yeah, this is a pattern for checking whether any member of the group matches the condition. I guess `TRUE > FALSE`.